### PR TITLE
cinnamon.updateScript: add updateScript

### PR DESCRIPTION
**Pull request by mkg20001**
###### Motivation for this change

An update script for cinnamon, to make stuff auto-update

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/cinnamon/gh_update/default.nix [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{}:
+
+let
+  pkgs = import ../../../../. { };
+  nixNodePackage = builtins.fetchGit {
```
Comment (mkg20001) [inline pkgs/desktops/cinnamon/gh_update/default.nix]: This could maybe be integrated into nixpkgs

**Comment by worldofpeace** [comment]
Do you still need this @mkg20001 ? The main change that needs to be made is it can't use IFD. I'd merge it right away after that

**Comment by mkg20001** [comment]
IFD?

**Comment by bobby285271** [comment]
> IFD?

I don't really know, maybe [this](https://github.com/NixOS/nixpkgs/blob/7a22a6a26f402d33121f834d1af87cb6eed75390/pkgs/desktops/cinnamon/gh_update/default.nix#L9) is a case of [IFD](https://nixos.wiki/wiki/Import_From_Derivation)? :thinking: 

P.s.: Nowadays there is `gitUpdater`, is it possible to use it (example in #181225)? I guess it should just work for most case [^1] with `ignoredVersions = "master*";`.

[^1]: Mint themes and icons [don't tag releases](https://github.com/linuxmint/mint-themes/issues/332), and I haven't found upstream git repo for [mint-artwork](http://packages.linuxmint.com/pool/main/m/mint-artwork/) yet.

**Comment by mkg20001** [comment]
@bobby285271 the problem with some cinnamon packages is that there are no tags, instead there are commits, etc.

basically lots of non-standard behaviour

---

this pr has been stale for quite a while so feel free to reuse / close

**Pull request closed without merging**